metpy/calc/kinematics.py

```diff
@@ -658,31 +659,50 @@ def potential_vorticity_baroclinic(potential_temperature, pressure, u, v, dx, dy
 
     Returns
     -------
-    (M, N) ndarray
+    (P, M, N) ndarray
         baroclinic potential vorticity
 
     Notes
     -----
-    The same formula is used for isobaric and isentropic PV analysis. Provide winds
-    for vorticity calculations on the desired isobaric or isentropic surface. Three layers
-    of pressure/potential temperature are required in order to calculate the vertical
-    derivative (one above and below the desired surface).
+    This function will only work with data that is in (P, Y, X) format. If your data
+    is in a different order you will need to re-order your data in order to get correct
+    results from this function.
+
+    The same function can be used for isobaric and isentropic PV analysis. Provide winds
+    for vorticity calculations on the desired isobaric or isentropic surface. At least three
+    layers of pressure/potential temperature are required in order to calculate the vertical
+    derivative (one above and below the desired surface). The first two terms will be zero if
+    isentropic level data is used due to the gradient of theta in both the x and y-directions
+    will be zero since you are on an isentropic surface.
+
+    This function expects pressure/isentropic level to increase with increasing array element
+    (e.g., from higher in the atmosphere to closer to the surface. If the pressure array is
+    one-dimensional p[:, None, None] can be used to make it appear multi-dimensional.)
 
     """
-    if np.shape(potential_temperature)[axis] != 3:
-        raise ValueError('Length of potential temperature along axis '
-                         '{} must be 3.'.format(axis))
-    if np.shape(pressure)[axis] != 3:
+    if np.shape(potential_temperature)[-3] < 3:
+        raise ValueError('Length of potential temperature along the pressure axis '
+                         '{} must be at least 3.'.format(-3))
+    if np.shape(pressure)[-3] < 3:
```

Comment:
These also have to be the same length right? Maybe 
```python
if (np.shape(potential_temperature)[-3] < 3) or (np.shape(pressure)[-3] < 3) or (np.shape(potential_temperature)[-3] != (np.shape(pressure)[-3] < 3)) : ValueError
```

Reply:
Yes, they would have to be the same length, so they could be combined in a more complex if statement.